### PR TITLE
coeff(): replace unnecessary RCP function args

### PR DESCRIPTION
**Pull request by rwst**
Depends on #1005 because of merge conflicts.

### CODE REVIEW

File: symengine/visitor.h [hunk 1]
```diff
@@ -101,8 +101,8 @@ bool has_symbol(const Basic &b, const Symbol& x);
 class CoeffVisitor : public BaseVisitor<CoeffVisitor, StopVisitor>
 {
 protected:
-    RCP<const Symbol> x_;
-    RCP<const Basic> n_;
+    const Symbol* x_;
+    const Basic* n_;
```
Comment (certik) [inline symengine/visitor.h]: Use `Ptr<const Symbol>` etc., instead of the raw pointers.

**Comment by certik** [comment]
Otherwise +1.

**Comment by rwst** [comment]
`Ptr` cannot have `nullptr` so it cannot have state. However, you have two distinct states with visitors: unapplied and applied. In other words, the `Ptr<const Symbol>` is very much like a reference that must be immediately initialized.

**Comment by rwst** [comment]
Closed in favor of #1018

**Pull request closed without merging**